--- src/game/controllers/user-games.controller.spec.ts
+++ src/game/controllers/user-games.controller.spec.ts
@@ -3,8 +3,8 @@
 import { GameService } from '../services/game.service';
 import { AuthGuard } from '../../auth/guards/auth.guard';
-import { Response } from 'express';
 import { AuthRequest } from '../../auth/interfaces/auth-user.interface';
+import { StreamableFile } from '@nestjs/common';

 describe('UserGamesController', () => {
   let controller: UserGamesController;
@@ -43,26 +43,15 @@
         .mockResolvedValue(mockCsvData);

       const request = { user: { sub: 'user1' } };
-      const response: Partial<Response> = {
-        setHeader: jest.fn(),
-        status: jest.fn().mockReturnThis(),
-        send: jest.fn(),
-      };

-      await controller.exportGames(
-        request as unknown as AuthRequest,
-        response as unknown as Response,
+      const result = await controller.exportGames(
+        request as unknown as AuthRequest
       );

       // eslint-disable-next-line @typescript-eslint/unbound-method
       expect(gameService.exportUserGamesToCsv).toHaveBeenCalledWith(
         request.user,
       );
-      expect(response.setHeader).toHaveBeenCalledWith(
-        'Content-Type',
-        'text/csv',
-      );
-      expect(response.setHeader).toHaveBeenCalledWith(
-        'Content-Disposition',
-        'attachment; filename="meu_historico_xadrez.csv"',
-      );
-      expect(response.status).toHaveBeenCalledWith(200);
-      expect(response.send).toHaveBeenCalledWith(mockCsvData);
+      expect(result).toBeInstanceOf(StreamableFile);
     });
   });
 });
